kore/src/Kore/Step/Simplification/NoConfusion.hs

```diff
@@ -27,6 +25,31 @@ import Prelude.Kore hiding (
     concat,
  )
 
+data UnifyEqualInjectiveHeadsAndEquals = UnifyEqualInjectiveHeadsAndEquals
+    { firstHead :: Symbol
+    , firstChildren :: [TermLike RewritingVariableName]
+    , secondChildren :: [TermLike RewritingVariableName]
+    }
+
+matchEqualInjectiveHeadsAndEquals ::
+    TermLike RewritingVariableName ->
+    TermLike RewritingVariableName ->
+    Maybe UnifyEqualInjectiveHeadsAndEquals
+matchEqualInjectiveHeadsAndEquals first second
+    | App_ firstHead firstChildren <- first
+      , App_ secondHead secondChildren <- second
+      , Symbol.isInjective firstHead
+      , Symbol.isInjective secondHead
+      , firstHead == secondHead --is one of the above redundant in light of this?
```

Comment:
You can remove `Symbol.isInjective secondHead` above, and leave a comment explaining.